### ISSUE TITLE
`docker service tasks --filter name=` filters on `service`

### ISSUE THREAD

**Issue by albers** [docker/docker#24112]
Related issue: #24108

According to https://github.com/docker/docker/blob/master/docs/reference/commandline/service_tasks.md#name 
>The name filter matches on task names.

This is not true: It filters on **exact service names**, which does not make sense because you also have to specify the service name as a mandatory argument.

```bash
# we have a service "top" with 3 tasks.
$ docker service tasks top
ID                         NAME   SERVICE  IMAGE    LAST STATE          DESIRED STATE  NODE
brscnd76m5eh8l3cvmpt5lkj4  top.1  top      busybox  Running 26 minutes  Running        bd4242800842
c4id3nmv05ivk1brv7qk5ik3e  top.2  top      busybox  Running 26 minutes  Running        04f850c4a3db
8t2qoxgr0e95w7bs0gwunf0az  top.3  top      busybox  Running 26 minutes  Running        04f850c4a3db

# This should find "top.2" but does not.
$ docker service tasks --filter name=top.2 top
ID  NAME  SERVICE  IMAGE  LAST STATE  DESIRED STATE  NODE

# Filtering on a substring won't do:
$ docker service tasks --filter name=to top
ID  NAME  SERVICE  IMAGE  LAST STATE  DESIRED STATE  NODE

# You have to specify the EXACT SERVICE name to find something.
$ docker service tasks --filter name=top top
ID                         NAME   SERVICE  IMAGE    LAST STATE          DESIRED STATE  NODE
brscnd76m5eh8l3cvmpt5lkj4  top.1  top      busybox  Running 27 minutes  Running        bd4242800842
c4id3nmv05ivk1brv7qk5ik3e  top.2  top      busybox  Running 27 minutes  Running        04f850c4a3db
8t2qoxgr0e95w7bs0gwunf0az  top.3  top      busybox  Running 27 minutes  Running        04f850c4a3db
```

**Comment by tiborvass** [comment]
Related issue #24152

**Comment by yongtang** [comment]
@albers @tiborvass I created a pull request #24620 to allow partial name match for `service ls --filter`. As part of the pull request, the task filter has also been updated. 

I think the `docker service tasks --filter name=` filters on `service` will be fixed in #24620.

Please let me know if there are any issues.
